mood-analysis.rb

```diff
@@ -19,6 +23,60 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def happy_days(words)
+  entries = 0
+  happy = 0
+  strip = []
+
+  words.each do |word|
+    strip << strip_punctuation(word)
+  end
+
+  strip.each do |string|
+    if happy < 3
+      entries += 1
+      if analyze_mood(string) == ":-)"
+        happy += 1
+      end
+    elsif happy == 3
+      puts "It takes #{entries} entries for 3 happy days to occur."
+    else
+      puts "There were less than 3 happy days in #{entries} entries. :-("
+    end
+  end
+end
+
+def overall_mood(words)
+  happy = 0
+  sad = 0
+  neutral = 0
+  strip = []
+
+  words.each do |word|
+    strip << strip_punctuation(word)
+  end
+
+  strip.each do |string|
+    if analyze_mood(string) == ":-)"
+      happy += 1
+    elsif analyze_mood(string) == ":-("
+      sad += 1
+    elsif analyze_mood(string) == ":-|"
+      neutral +=0
+    end
+  end
+
+  if happy > (sad && neutral)
+    puts "The most common mood is :-)"
+  elsif sad > (happy && neutral)
+    puts "The most common mood is :-("
+  elsif neutral > (sad && happy)
+    puts "The most common mood is :-|"
+  else
+    puts "The most common mood is unable to be determined."
+  end
+end
```

Comment:
Then here we'd be able to utilize the hash's methods to find the max mood easier as well.